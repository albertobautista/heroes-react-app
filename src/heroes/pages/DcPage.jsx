import React from "react";
import { HeroesList } from "../components";

export const DcPage = () => {
  return (
    <>
      <h1>DC Page</h1>
      <HeroesList publisher="DC Comics" />
    </>
  );
};
