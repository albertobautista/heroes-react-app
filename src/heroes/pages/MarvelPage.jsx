import React from "react";
import { HeroesList } from "../components";

export const MarvelPage = () => {
  return (
    <>
      <h1>Marvel Page</h1>
      <HeroesList publisher="Marvel Comics" />
    </>
  );
};
