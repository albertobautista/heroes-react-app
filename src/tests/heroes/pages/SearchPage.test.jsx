import { render, screen } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import { SearchPage } from "../../../heroes/pages/SearchPage";

describe("Test in SearchPage", () => {
  test("Should show correctly with default values", () => {
    const { container } = render(
      <MemoryRouter>
        <SearchPage />
      </MemoryRouter>
    );
    expect(container).toMatchSnapshot();
  });

  test("Should show batman and input with query string value", () => {
    render(
      <MemoryRouter initialEntries={["/search?q=batman"]}>
        <SearchPage />
      </MemoryRouter>
    );

    const input = screen.getByRole("textbox");
    const img = screen.getByRole("img");

    expect(input.value).toBe("batman");
    expect(img.src).toContain("/assets/heroes/dc-batman.jpg");
  });
});
