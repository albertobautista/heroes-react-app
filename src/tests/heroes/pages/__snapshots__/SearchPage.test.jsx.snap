// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in SearchPage Should show batman and input with query string value 1`] = `
<div>
  <h1>
    Search
  </h1>
  <hr />
  <div
    class="row"
  >
    <div
      class="col-5"
    >
      <h4>
        Searching
      </h4>
      <hr />
      <form>
        <input
          autocomplete="off"
          class="form-control"
          name="searchText"
          placeholder="Search a hero.."
          type="text"
          value=""
        />
        <button
          class="btn btn-outline-primary mt-2"
          type="submit"
        >
          Search
        </button>
      </form>
    </div>
    <div
      class="col-7"
    >
      <h4>
        Results
      </h4>
      <hr />
      <div
        class="alert alert-primary"
      >
        Search a hero
      </div>
    </div>
  </div>
</div>
`;

exports[`Test in SearchPage Should show correctly with default values 1`] = `
<div>
  <h1>
    Search
  </h1>
  <hr />
  <div
    class="row"
  >
    <div
      class="col-5"
    >
      <h4>
        Searching
      </h4>
      <hr />
      <form>
        <input
          autocomplete="off"
          class="form-control"
          name="searchText"
          placeholder="Search a hero.."
          type="text"
          value=""
        />
        <button
          class="btn btn-outline-primary mt-2"
          type="submit"
        >
          Search
        </button>
      </form>
    </div>
    <div
      class="col-7"
    >
      <h4>
        Results
      </h4>
      <hr />
      <div
        class="alert alert-primary"
      >
        Search a hero
      </div>
    </div>
  </div>
</div>
`;
